ms-common-impl/src/main/java/net/trajano/ms/vertx/beans/JwksProvider.java

```diff
@@ -45,16 +41,22 @@
 
     public static final int MIN_NUMBER_OF_KEYS = 2;
 
+    @Autowired(required = false)
+    private CacheManager cm;
     /**
      * This is a cache of JWKs. If this is not provided a default one is used.
      */
     private Cache jwksCache;
 
     private KeyPairGenerator keyPairGenerator;
 
-    private Random random;
+    /**
+     * Random source.  Use ThreadLocalRandom since these will not be used for secure keys.
+     */
+    private Random random = ThreadLocalRandom.current();
 
-    private TokenGenerator tokenGenerator;
+    @Autowired
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "random" private field. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1068)